source/MaterialXView/Viewer.cpp

```diff
@@ -519,6 +513,50 @@ void Viewer::createLoadMaterialsInterface(Widget* parent, const std::string& lab
     });
 }
 
+void Viewer::createSaveMaterialsInterface(Widget* parent, const std::string& label)
+{
+    ng::Button* materialButton = new ng::Button(parent, label);
+    materialButton->setIcon(ENTYPO_ICON_SAVE);
+    materialButton->setCallback([this]()
+    {
+        mProcessEvents = false;
+        std::string filename = ng::file_dialog({ { "mtlx", "MaterialX" } }, true);
+
+        //sync document with editor changes
+        for(auto& material: _materials)
+        {
+            material->persistUniformChanges();
+        }
+
+        //save document
+        if (!filename.empty() && !_materials.empty())
+        {
+            mx::DocumentPtr doc = _materials.front()->getDocument();
```

Comment:
This a document which is cached on the Viewer, so no need to check the material for the document .

Reply:
The only document that I saw was `_stdLib` but I guess it's for standard libraries, do you mean another cached doc? Thanks
```
    // Document management
    mx::StringVec _libraryFolders;
    mx::FileSearchPath _searchPath;
    mx::DocumentPtr _stdLib;
    mx::FilePath _materialFilename;
    DocumentModifiers _modifiers;
```

Reply:
Sorry about that, I have a local branch where I cached the document as a data member called _materialDocument. And assign it in Viewer::loadDocument() where these lines are:
```
     // Load source document.
        mx::DocumentPtr doc = mx::createDocument();
```
i.e. instead of a local doc variable, I changed it to _materialDocument.